src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,109 @@
-import { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie:React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [formData, setFormData] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [submitButton, setSubmitButton] = useState(true);
```

Comment:
submitButton it's sound as text for submit button
I suggest you to rename it to isSubmitButtonDisabled
```suggestion
  const [isSubmitButtonDisabled, setIsSubmitButtonDisabled] = useState(true);
```